src/http/request.rs

```diff
@@ -111,6 +111,15 @@ impl Request {
         self.0.connection
     }
 
+    /// Pointer to a [`ngx_http_upstream_t`] upstream server object.
+    ///
+    /// [`ngx_http_upstream_t`]: is best described in
+    /// https://nginx.org/en/docs/dev/development_guide.html#http_request
+    /// https://nginx.org/en/docs/dev/development_guide.html#http_load_balancing
+    pub fn upstream(&self) -> *mut ngx_http_upstream_t {
```

Comment:
Is it more idiomatic to return Option<*mut T> here?

And allows callers to Option.unwrap() and `match` against Some / None?

Reply:
Returning the pointer is a patterned followed from other methods, meaning, it's consistent with access to other ngx_http_request_t fields. But should Option be used for all methods?